_posts/2018-03-08-roadmap.md

```diff
@@ -0,0 +1,221 @@
+[stability]: https://blog.rust-lang.org/2014/10/30/Stability.html
+
+Epochs will have a meaning for the compiler. You will be able to write:
+
+```toml
+epoch = "2018"
+```
+
+in your Cargo.toml to _opt in_ to the new epoch for your crate. Doing so may
+introduce new keywords or otherwise require adjustments to code. However:
+
+-   You can use _old_ epochs indefinitely on _new_ compilers; **epochs are
+    opt-in**.
+-   Epochs are set on a _per-crate_ basis and can be mixed and matched; **you
+    can be on a different epoch from your dependencies**.
+-   Warning-free code in one epoch must compile, and have the same behavior, on
+    the next.
+-   Epoch-related warnings, e.g. that an identifier will become a keyword in the
+    next epoch, must be easily fixable via an automated migration tool
+    (rustfix). **Only a small minority of crates should require** **_any_**
+    **manual work to opt in to a new** epoch, and that manual work must be
+    minimal.
+
+In other words, the progression of new compiler versions is independent from
+epoch; you can migrate at your leisure, and don’t have to worry about ecosystem
+compatibility; and epoch migration is normally trivial.
+
+## What will be in Rust 2018?
+
+The roadmap doesn’t say _for certain_ what will ship in Rust 2018, but we have a
+pretty good idea, and we’ll cover the major suspects below.
+
+### Documentation improvements
+
+Part of the goal with the Rust 2018 release is to provide high quality
+documentation for the full set of new and improved features and the idioms they
+give rise to. [The Rust Programming Language book][trpl] has been completely
+re-written over the last 18 months, and will be updated throughout the year as
+features reach the stable compiler. [Rust By Example] will likewise undergo a
+revamp this year. And there are numerous third party books, like [Programming
+Rust], reaching print as well.
+
+[trpl]: https://doc.rust-lang.org/nightly/book/second-edition/
+[Programming Rust]: https://www.google.com/search?q=programming+rust&ie=utf-8&oe=utf-8&client=firefox-b-1-ab
+[Rust By Example]: https://rustbyexample.com/
+
+**Language improvements**
+
+The most prominent language work in the pipeline stems from [2017’s ergonomics
+initiative]. Almost all of the accepted RFCs from the initiative are available
+on nightly today, and will be polished and stabilized over the next several
+months. Among these productivity improvements are a few “headliners” that will
+form the backbone of the release:
+
+[2017’s ergonomics initiative]: https://blog.rust-lang.org/2017/03/02/lang-ergonomics.html
+
+-   **Ownership system improvements**, including making borrowing more flexible
+    via “non-lexical lifetimes”, improved pattern matching integration, and
+    more.
+-   **Trait system improvements**, including the long-awaited impl Trait syntax
+    for dealing with types abstractly.
+-   **Module system improvements**, focused on increasingly clarity and reducing
+    complexity.
```

Comment:
The non-marketing version: Experimentation with module system is happening, nobody knows how it will affect learnability, clarity and complexity.